python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py

```diff
@@ -108,3 +108,72 @@ def test_basic_resource_request(
             poll_timeout=2,
             poll_interval=1,
         )
+
+
+@pytest.mark.parametrize(
+    "n_polls, succeed_at_end",
+    [(0, True), (0, False), (4, True), (4, False), (30, True)],
+)
+def test_sync_and_poll(n_polls, succeed_at_end, connector_id):
+    resource = FivetranWorkspace(
+        account_id=TEST_ACCOUNT_ID, api_key=TEST_API_KEY, api_secret=TEST_API_SECRET
+    )
+    client = resource.get_client()
+
+    test_connector_api_url = get_fivetran_connector_api_url(connector_id)
+
+    test_succeeded_at = TEST_MAX_TIME_STR
+    test_failed_at = TEST_PREVIOUS_MAX_TIME_STR
+    # Set `failed_at` as more recent that `succeeded_at` if the sync and poll process is expected to fail
+    if not succeed_at_end:
+        test_succeeded_at, test_failed_at = test_failed_at, test_succeeded_at
```

Comment:
this is kind of a weird way to do this but I think it's fine. Just add a comment

Reply:
I changed it in [05442c2](https://github.com/dagster-io/dagster/pull/26061/commits/05442c2eb793989ba4d8792280489c42c4272da6) to make it clearer